### PR TITLE
Add deploy script for Flowers

### PR DESCRIPTION
**Pull request by nambrot**

### CODE REVIEW

File: scripts/deploy.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      name: 'Flow3rs 2022',
+      symbol: 'FLOW3',
+      totalSupply: 0,
+      owner: signer.address,
+      mailbox: '0x1d3aAC239538e6F1831C8708803e61A9EA299Eec',
+      interchainGasPaymaster: core.getContracts(Chains.celo)
+        .interchainGasPaymaster.address,
+    } as SyntheticConfig & RouterConfig,
+    ethereum: {
+      type: 'COLLATERAL',
```
Comment (yorhodes) [inline scripts/deploy.ts:52]:
```suggestion
      type: 'SYNTHETIC_URI',
      name: 'Flow3rs 2022',
      symbol: 'FLOW3',
      totalSupply: 0,
      owner: signer.address,
      mailbox: '0x1d3aAC239538e6F1831C8708803e61A9EA299Eec',
      interchainGasPaymaster: core.getContracts(Chains.celo)
        .interchainGasPaymaster.address,
    } as SyntheticConfig & RouterConfig,
    ethereum: {
      type: 'COLLATERAL_URI',
```

**Review by yorhodes — CHANGES_REQUESTED** [review]

File: scripts/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  console.info('Preparing utilities');
+  const chainProviders = objMap(prodConfigs, (_, config) => ({
+    provider: config.provider,
+    confirmations: config.confirmations,
+    overrides: config.overrides,
+    signer: new Wallet(
+      'pkey',
+      config.provider,
+    ),
+  }));
```
[the remark this answers is not in the document]
Comment (nambrot) [reply scripts/deploy.ts:34]: We actually do hav `multiProvider.rotateSigner` 😅

File: scripts/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    celo: {
+      type: 'SYNTHETIC_URI',
+      name: 'Flow3rs 2022',
+      symbol: 'FLOW3',
+      totalSupply: 0,
+      owner: signer.address,
+      mailbox: '0x1d3aAC239538e6F1831C8708803e61A9EA299Eec',
+      interchainGasPaymaster: core.getContracts(Chains.celo)
+        .interchainGasPaymaster.address,
+    } as SyntheticConfig & RouterConfig,
```
[the remark this answers is not in the document]
Comment (nambrot) [reply scripts/deploy.ts:49]: Right now we do, cause we don't specify an ISM (which is what the type requires)

File: scripts/deploy.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  const signer = new Wallet(
+    'pkey',
+  );
+
+  const multiProvider = new MultiProvider(connectionConfigs);
+  multiProvider.rotateSigner(signer)
+  const core = HyperlaneCore.fromEnvironment('mainnet', multiProvider);
+  const config = {
+    celo: {
+      type: 'SYNTHETIC_URI',
```
Comment (jmrossy) [inline scripts/deploy.ts:30]: Should this be just SYNTHETIC to match the const in config?
https://github.com/hyperlane-xyz/hyperlane-token/blob/main/src/config.ts#L6

Also, would be more conventional to have these be enums, whose values would typically be lowercase. cc @yorhodes

File: scripts/deploy.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      name: 'Flow3rs 2022',
+      symbol: 'FLOW3',
+      totalSupply: 0,
+      owner: signer.address,
+      mailbox: '0x1d3aAC239538e6F1831C8708803e61A9EA299Eec',
```
Comment (jmrossy) [inline scripts/deploy.ts:35]: Should this come from the SDK?

File: contracts/HypERC721.sol [hunk 6]
```diff
@@ -14,24 +14,21 @@ contract HypERC721 is ERC721EnumerableUpgradeable, TokenRouter {
...
-        address _interchainSecurityModule,
         uint256 _mintAmount,
         string memory _name,
         string memory _symbol
     ) external initializer {
         // transfers ownership to `msg.sender`
         __HyperlaneConnectionClient_initialize(
             _mailbox,
-            _interchainGasPaymaster,
-            _interchainSecurityModule
```
Comment (jmrossy) [inline contracts/HypERC721.sol:34]: Are we cutting the ISM concept for v0?

File: scripts/deploy.ts (hunk 5 above)
Comment (yorhodes) [reply scripts/deploy.ts:35]: I think because we don't have v2 deployed on mainnet he did this

File: contracts/HypERC721.sol (hunk 6 above)
Comment (yorhodes) [reply contracts/HypERC721.sol:34]: good question

Comment (yorhodes) [reply contracts/HypERC721.sol:34]: I think he did this for the deploy but not sure we actually want to publish it this way

Comment (nambrot) [reply contracts/HypERC721.sol:34]: No, but currently the router asserts that you can't initialize this to an address that is not a contract which means 0x0 is not valid. IMO we should have multiple initializers and not require the passing of ISMs if the default is to be used

Comment (nambrot) [reply contracts/HypERC721.sol:34]: Users can always set via the explicit setISM function

**Pull request merged**